commit bb0d04663c7dc6c0096f8717cb4ec26330a5ae40
Author: Yun Peng <pcloudy@google.com>
Date:   Wed Jun 3 15:35:31 2020 +0200

    Patch grpc v1.26.0 for Bazel build
    
    - Avoid loading dependencies that're not needed for the gRPC C++
    libraries
    - Redirect zlib to @//third_party/zlib

diff --git a/bazel/grpc_build_system.bzl b/bazel/grpc_build_system.bzl
index 7bb6b8bdb9..7644107b70 100644
--- a/bazel/grpc_build_system.bzl
+++ b/bazel/grpc_build_system.bzl
@@ -25,7 +25,7 @@
 load("//bazel:cc_grpc_library.bzl", "cc_grpc_library")
 load("//bazel:copts.bzl", "GRPC_DEFAULT_COPTS")
 load("@upb//bazel:upb_proto_library.bzl", "upb_proto_library")
-load("@build_bazel_rules_apple//apple:ios.bzl", "ios_unit_test")
+# load("@build_bazel_rules_apple//apple:ios.bzl", "ios_unit_test")
 
 # The set of pollers to test against if a test exercises polling
 POLLERS = ["epollex", "epoll1", "poll"]
@@ -181,13 +181,13 @@ def ios_cc_test(
             testonly = 1,
         )
         ios_test_deps = [ios_test_adapter, ":" + test_lib_ios]
-        ios_unit_test(
-            name = name + "_on_ios",
-            size = kwargs.get("size"),
-            tags = ios_tags,
-            minimum_os_version = "9.0",
-            deps = ios_test_deps,
-        )
+        # ios_unit_test(
+        #     name = name + "_on_ios",
+        #     size = kwargs.get("size"),
+        #     tags = ios_tags,
+        #     minimum_os_version = "9.0",
+        #     deps = ios_test_deps,
+        # )

 def grpc_cc_test(name, srcs = [], deps = [], external_deps = [], args = [], data = [], uses_polling = True, language = "C++", size = "medium", timeout = None, tags = [], exec_compatible_with = [], exec_properties = {}, shard_count = None, flaky = None, copts = []):
     copts = copts + if_mac(["-DGRPC_CFSTREAM"])
diff --git a/bazel/grpc_deps.bzl b/bazel/grpc_deps.bzl
index 09fcad95a2..9b737e5deb 100644
--- a/bazel/grpc_deps.bzl
+++ b/bazel/grpc_deps.bzl
@@ -33,7 +33,7 @@ def grpc_deps():
 
     native.bind(
         name = "madler_zlib",
-        actual = "@zlib//:zlib",
+        actual = "@//third_party/zlib",
     )
 
     native.bind(
diff --git a/bazel/grpc_extra_deps.bzl b/bazel/grpc_extra_deps.bzl
index 09fcad95a2..9b737e5deb 100644
--- a/bazel/grpc_extra_deps.bzl
+++ b/bazel/grpc_extra_deps.bzl
@@ -15,11 +15,6 @@
 
 load("@com_google_protobuf//:protobuf_deps.bzl", "protobuf_deps")
 load("@upb//bazel:workspace_deps.bzl", "upb_deps")
-load("@envoy_api//bazel:repositories.bzl", "api_dependencies")
-load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
-load("@build_bazel_rules_apple//apple:repositories.bzl", "apple_rules_dependencies")
-load("@build_bazel_apple_support//lib:repositories.bzl", "apple_support_dependencies")
-load("@com_google_googleapis//:repository_rules.bzl", "switched_rules_by_language")
 
 def grpc_extra_deps(ignore_version_differences = False):
     """Loads the extra dependencies.
@@ -43,20 +38,3 @@
     protobuf_deps()
 
     upb_deps()
-
-    api_dependencies()
-
-    go_rules_dependencies()
-    go_register_toolchains()
-
-    apple_rules_dependencies(ignore_version_differences = ignore_version_differences)
-
-    apple_support_dependencies()
-
-    # Initialize Google APIs with only C++ and Python targets
-    switched_rules_by_language(
-        name = "com_google_googleapis_imports",
-        cc = True,
-        grpc = True,
-        python = True,
-    )
